Source Main 5.2/source/NewUIMyInventory.cpp

```diff
@@ -328,8 +328,8 @@ bool CNewUIMyInventory::IsEquipable(int iIndex, ITEM* pItem)
 
     if (pItem->Type == ITEM_DARK_RAVEN_ITEM)
     {
-        const PET_INFO* pPetInfo = GetPetInfo(pItem);
-        const WORD wRequireCharisma = (185 + (pPetInfo->m_wLevel * 15));
+        auto pPetInfo = GetPetInfo(pItem);
+        auto wRequireCharisma = (185 + (pPetInfo->m_wLevel * 15));
```

Comment:
WORD can't hold the computed data. changed to auto instead.

Reply:
WORD can hold the required charisma. The pet level is usually capped at level 50, so the result is nowhere near 65k. The root of the issue is probably something else

Reply:
in this code block it only checks whether the charisma is enough or not, returns boolean. i think this should enough for this particular code block?

Reply:
yeah, rethinking about that formula, i get the point that it should not exceed 65k

Reply:
The pet data is somehow not initialized:
![grafik](https://github.com/user-attachments/assets/51e9fb7e-294b-4af3-b5de-65ed3b19c543)

Reply:
I think this is due the increased frame rate... I get this uninitialized data for two frames, until the pet info is received from the server. However, with this code, the intent should be clearer:
```
    if (pItem->Type == ITEM_DARK_RAVEN_ITEM)
    {
        const auto pPetInfo = GetPetInfo(pItem);
        if (pPetInfo->m_dwPetType == PET_TYPE_NONE)
        {
            return false;
        }

        const auto requiredCharisma = (185 + (pPetInfo->m_wLevel * 15));
        if (requiredCharisma > wCharisma)
        {
            return false;
        }
    }
```

Reply:
I'll apply this suggestion. Many thanks!

Reply:
The server is sending this data on-demand for non-equipped raven/horses. As soon as you hover over the item with your mouse pointer, it sends a packet which requests this info. I think that with the lower frame rate, the response from the server was fast enough to be available at the next frame.

Reply:
Related: https://github.com/sven-n/MuMain/issues/12

Reply:
Applied in latest commit. Kindly check. Thanks!